### ISSUE TITLE
.gitmodules references

### ISSUE THREAD

**Issue by FirefoxMetzger** [thirddegree/Hatchit#20]
When forking or cloning the repository, the submodules will point to the same modules as in this repository. Many of the ThirdParty subs are pointing to forks of the original repo held by `thirddegree`. 

If the reference isn't necessary (special modifications) wouldn't it make more sense to have them point to the 'official' repositories? As they point to specific commits they are uncorrelated with potential unstable HEADs of those repos.

The reason I am asking is, because I don't know how long lived the ThirdParty repos are. For example in the `pre-build` branch is a link to some `python` sub, which is held by `thirddegree`. This one is either private or deleted and causes some annoyance when trying to build Hatchit for Windows (It is the branch with the `Windows` folder described in the README ... at least I think it is :) )

**Comment by MattGuerrette** [comment]
The python related third-party repos will most likely be removed soon. Yes, I agree that the third module libraries should point to the official repo, and they do for some. But, for a few I forked them so I could implement a CMake build as to integrate more nicely with our build system for Hatchit.

Also, currently the build system for Hatchit is undergoing some serious restructuring and has only been redone on Linux at the moment. I still need to refactor the Windows build, and until then it will be unable to be built.

I should have a working Windows build up by the end of the day tomorrow. So if you are looking to test out the engine or become a contributing developer let me know.

**Comment by FirefoxMetzger** [comment]
I tried exactly the same with CMakeLists in different projects a while back, however it didn't really work out for me. Recently I found a neat 'hack' using find modules to create imported targets with transitive dependencies. The advantage is a clean dependency repo and a `.cmake` rather robust to updates of the dependency. The disadvantage is that it forces me to build the dependencies seperately. For deployment (I think thats the right technical term) with CMake I am currently experimenting with having the parent CMake create isolated generator targets for each dependency if they also use CMake. However, this is not very battle tested yet.

I saw the enhancement on the issue list, but as it was from 2 month ago, so I figured that linux development is currently on a higher priority. Thus, I decided to start reworking the CMakeLists for Windows myself to get a build running that passes the tests.

I definitly want to check and test the engine, but I will be moving by the end of the week and then again in the middle of the month and I don't know how good internet access will be during that time. So for now it is 'teach myself some vulkan and more graphics programming'.
